src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,278 @@
+    }
+  }
+
+  disconnectedCallback() {
+    this.disable();
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.unassociateElement?.(this);
+    }
+  }
+
+  get keysUsed() {
+    return ['Enter', ' ', 'ArrowDown', 'ArrowUp', 'Home', 'End'];
+  }
+
+  #getItem(e) {
+    if (e.target === this) return;
+
+    const composedPath = e.composedPath();
+    const index = composedPath.findIndex(el => el.nodeName === 'MEDIA-CHROME-LISTITEM');
+
+    return composedPath[index];
+  }
+
+  handleSelection(e) {
+    const item = this.#getItem(e);
+
+    if (!item) return;
+
+    const selected = item.getAttribute('aria-selected') === 'true';
+
+    if (this.getAttribute('aria-multiselectable') !== 'true') {
+      this.#assignedElements.forEach(el => el.setAttribute('aria-selected', 'false'));
+    }
+
+    if (selected) {
+      item.setAttribute('aria-selected', 'false');
+    } else {
+      item.setAttribute('aria-selected', 'true');
+    }
+  }
+
+  handleMovement(e) {
+    const { key } = e;
+    const els = this.#items;
+
+    let currentOption = this.#getItem(e);
+    if (!currentOption) {
+      currentOption = els.filter(el => el.getAttribute('tabindex') === '0')[0];
+    }
+
+    let nextOption;
+    
+    switch (key) {
+      case 'ArrowDown':
+        nextOption = currentOption.nextElementSibling;
+
+        if (nextOption.hasAttribute('disabled')) {
+          nextOption = nextOption.nextElementSibling;
+        }
+
+        break;
+      case 'ArrowUp':
+        nextOption = currentOption.previousElementSibling;
+
+        if (nextOption.hasAttribute('disabled')) {
+          nextOption = nextOption.previousElementSibling;
+        }
+
+        break;
+      case 'Home':
+        nextOption = els[0];
+        break;
+      case 'End':
+        nextOption = els[els.length - 1];
+        break;
+      default:
+        nextOption = this.#searchItem(key);
+        break;
+    }
+
+    if (nextOption) {
+      els.forEach(el => el.setAttribute('tabindex', '-1'));
+      nextOption.setAttribute('tabindex', '0');
+      nextOption.focus();
+    }
+  }
+
+  handleClick(e) {
+    const item = this.#getItem(e);
+
+    if (!item || item.hasAttribute('disabled')) return;
+
+    this.#items.forEach(el => el.setAttribute('tabindex', '-1'));
+    item.setAttribute('tabindex', '0');
+
+    this.handleSelection(e);
+  }
+
+  #searchItem(key) {
+    this.#clearKeysOnDelay();
+
+    const els = this.#items;
+    const activeIndex = els.findIndex(el => el.getAttribute('tabindex') === '0');
+    const repeatedKey = this.#keysSoFar === key || this.#keysSoFar.length === 0;
+
+    // don't accumulate keys if it's a repeated key
+    // unless we haven't started accumulating a key yet
+    if (!repeatedKey || this.#keysSoFar.length === 0) {
```

Comment:
question(blocking): Does this mean it's invalid to search for, e.g. "oo"? Should it be (honest question)?

Reply:
yes, with the current implementation, it isn't possible. Off the top of my head, not sure how to support both.

Reply:
It looks like the reference implementation for listbox doesn't treat repeated keys as a special case:
https://codepen.io/pen?editors=0010
(referenced here: https://www.w3.org/WAI/ARIA/apg/example-index/listbox/listbox-scrollable.html)

Reply:
Let me see if I can get both working. I personally find the repeated keys thing super useful. If it's not a lot of code and doesn't take too long to implement, I'll do it, otherwise I'll take it out.

Reply:
I played around with how Windows Explorer behaves in this case, and I think I have a way forward here.
Basically, you'd still want to accumulate the keys, but if it's the same letter, just treat it as a single key unless there's a better match. Shouldn't be too bad to add in.

Reply:
Pushed a change for this